tests/unit/async_db/test_connection.py

```diff
@@ -90,18 +90,60 @@ async def test_cursor_initialized(
 
 @mark.asyncio
 async def test_connect_empty_parameters():
-    params = ("database", "username", "password")
-    kwargs = {"engine_url": "engine_url", **{p: p for p in params}}
-
-    for param in params:
-        with raises(InterfaceError) as exc_info:
-            kwargs = {
-                "engine_url": "engine_url",
-                **{p: p for p in params if p != param},
-            }
-            async with await connect(**kwargs):
-                pass
-        assert str(exc_info.value) == f"{param} is required to connect."
+    with raises(InterfaceError) as exc_info:
+        async with await connect(engine_url="engine_url"):
+            pass
+    assert str(exc_info.value) == "database name is required to connect."
+
+
+@mark.asyncio
+async def test_connect_access_token(
+    settings: Settings,
+    db_name: str,
+    httpx_mock: HTTPXMock,
+    auth_callback: Callable,
+    auth_url: str,
+    check_token_callback: Callable,
+    query_url: str,
+    python_query_data: List[List[ColType]],
+    access_token: str,
+):
+    httpx_mock.add_callback(check_token_callback, url=query_url)
+    async with (
+        await connect(
+            engine_url=settings.server,
+            database=db_name,
+            access_token=access_token,
+            account_name="a",
+            api_endpoint=settings.server,
+        )
+    ) as connection:
+        cursor = connection.cursor()
+        assert await cursor.execute("select*") == -1
+
+    with raises(InterfaceError) as exc_info:
+        async with await connect(engine_url="engine_url", database="database"):
+            pass
+    assert (
+        str(exc_info.value)
+        == "Neither username/password nor access_token are provided. Provide one to"
+        " authenticate"
+    )
```

Comment:
I'm not a fan of checking the exception strings. As long as the correct exception class is raised we should be ok. This way if we were to fix typos or reword the exceptions we wouldn't have to change tests.
In this case I'd actually create a new error class, something like SettingsError, since from the description InterfaceError is not exactly what we want here.